### PR TITLE
add search docs with metadata filter

### PR DESCRIPTION
**Pull request by carycheng**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @carycheng, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: docs/search.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+Search for Content
+------------------
+
+To get a list of items matching a serch query, call [`search.query(query, limit=None, offset=0, **kwargs)`][query] will return an `Iterable` that allows you 
+to iterate over the [`Item`][item_class] objects in the collection.
+
+```python
+items = client.search().query(query='TEST QUERY', limit=100, file_extensions=['pdf', 'doc'])
+for item in items:
+    print('The item ID is {} and the item name is {1}'.format(item.id, item.name)
```
Comment (mattwiller) [inline docs/search.md]: Missing the closing paren for `print()` here.  Also, both placeholders should be numbered

File: docs/search.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+[query]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.search.Search.query
+[item_class]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.item.Item
+
+### Metadata Search
+
+To filter by metadata in your search, first create [MetadataSearchFilter][metadata_search_filter] object with the
```
Comment (mattwiller) [inline docs/search.md]: Missing backticks around the `MetadataSearchFilter` class name

File: docs/search.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+[query]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.search.Search.query
+[item_class]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.item.Item
+
+### Metadata Search
+
+To filter by metadata in your search, first create [MetadataSearchFilter][metadata_search_filter] object with the 
+specified `template_key` and `scope` as well as adding filter, `field_key` and `value` with 
+[`metadata_search_filter.add_value_based_filter(field_key, value)`][add_value_based_filter]. Next, create a
```
Comment (mattwiller) [inline docs/search.md]: Consider shortening the object name here to `metadata_filter` to align with the next line where it's passed into `add_filter()`

File: docs/search.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+[query]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.search.Search.query
+[item_class]: https://box-python-sdk.readthedocs.io/en/latest/boxsdk.object.html#boxsdk.object.item.Item
+
+### Metadata Search
+
+To filter by metadata in your search, first create [MetadataSearchFilter][metadata_search_filter] object with the 
+specified `template_key` and `scope` as well as adding filter, `field_key` and `value` with 
+[`metadata_search_filter.add_value_based_filter(field_key, value)`][add_value_based_filter]. Next, create a 
+[`MetadataSearchFilters`][metadata_search_filters] object and call [`metadata_search_filters.add_filter(metadata_filter)`][add_filter] 
+and pass in the [MetadataSearchFilter][metadata_search_filter] object created earlier. Finally, call
```
Comment (mattwiller) [inline docs/search.md]: Missing backticks around `MetadataSearchFilter`

File: docs/search.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+To filter by metadata in your search, first create [MetadataSearchFilter][metadata_search_filter] object with the 
+specified `template_key` and `scope` as well as adding filter, `field_key` and `value` with 
+[`metadata_search_filter.add_value_based_filter(field_key, value)`][add_value_based_filter]. Next, create a 
+[`MetadataSearchFilters`][metadata_search_filters] object and call [`metadata_search_filters.add_filter(metadata_filter)`][add_filter] 
+and pass in the [MetadataSearchFilter][metadata_search_filter] object created earlier. Finally, call 
+[`search.query(query, metadata_filters=None, **kwargs)`][query] with [`MetadataSearchFilters`][metadata_search_filters] 
+object passed in.
+
+```python
+from boxsdk.object.search import MetadataSearchFilter, MetadataSearchFilters
```
Comment (mattwiller) [inline docs/search.md]: Consider adding a line break below this line to make the example more readable

File: docs/search.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+metadata_search_filter = MetadataSearchFilter(template_key='marketingCollateral', scope='enterprise')
+metadata_search_filter.add_value_based_filter(field_key='documentType', value='datasheet')
+metadata_search_filter.add_value_based_filter(field_key='clientNumber', value='a123')
+metadata_search_filters = MetadataSearchFilters()
+metadata_search_filters.add_filter(metadata_search_filter)
```
Comment (mattwiller) [inline docs/search.md]: Consider adding a line break below this line to make the example more readable

**Pull request merged**